Tools/gulp/gulp-nuspec-sync.js

```diff
@@ -0,0 +1,63 @@
+// through2 is a thin wrapper around node transform streams
+var through = require('through2');
+var gutil = require('gulp-util');
+var xml2js = require('xml2js');
+var path = require('path');
+var fs = require('fs');
+var PluginError = gutil.PluginError;
+
+const PLUGIN_NAME = 'gulp-nuspec-sync';
+
+function gulpNuspecSync(opts) {
+  // Creating a stream through which each file will pass
+  return through.obj(function(file, enc, cb) {
+    var pkgConfigPath = path.join(path.dirname(file.path), 'packages.config');
+    var pkgConfigString = fs.readFileSync(pkgConfigPath).toString();
+    xml2js.parseString(pkgConfigString, function(err, pkgConfigXml) {
+      xml2js.parseString(file.contents.toString(), function(err, nuspecXml) {
+
+        gutil.log('Syncing .nuspec and packages.config for: ' + file.path + ' and ' + pkgConfigPath);
+
+        var pkgVersions = {};
+        var packages = pkgConfigXml['packages']['package'];
+        packages.forEach(function(package) {
+          pkgVersions[package.$.id] = package.$.version;
+        });
+
+        nuspecXml['package']['metadata'][0]['dependencies'].forEach(function(dependencies) {
+          var updateDep = function(depend) {
+            depend.forEach(function(dep) {
+              if (dep.$.id in pkgVersions) {
+                if(pkgVersions[dep.$.id] != dep.$.version){
+                  gutil.log('Updating dependency: ' + dep.$.id + ' from: ' + pkgVersions[dep.$.id] + ' to: ' + dep.$.version)
```

Comment:
i read twice and still do not get what 'dep.$.id' means exactly, guess a new thing that i should catch up with. But i am a little certain that you should switch the order that should be  "from dep.$.version to pkgVersions[dep.$.id]"?